pkg/util/ip_reservation_test.go

```diff
@@ -0,0 +1,335 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"testing"
+)
+
+func initTestIPAllocator() *IPAllocator {
+	pendingIPRanges := make(map[string]bool)
+	return &IPAllocator{
+		pendingIPRanges: pendingIPRanges,
+	}
+}
+
+func TestValidateCIDR(t *testing.T) {
+	cases := []struct {
+		name     string
+		cidr     string
+		expected bool
+	}{
+		{
+			name:     "Valid /29 CIDR format",
+			cidr:     "192.168.92.192/29",
+			expected: true,
+		},
+		{
+			name:     "Invalid CIDR format",
+			cidr:     "192.168.92.192",
+			expected: false,
+		},
+		{
+			name:     "Invalid CIDR format with IP range overflow",
```

Comment:
nit: the reason is because of misalignment, not overflow.  maybe something like 245 could illustrate that better